flexget/plugins/input/anidb_list.py

```diff
@@ -26,69 +28,100 @@ class AnidbList(object):
                 'error_pattern': 'user_id must be in the form XXXXXXX'},
             'type': {
                 'type': 'string',
-                'enum': ['shows', 'movies'],
+                'enum': ['shows', 'movies', 'ovas'],
                 'default': 'movies'},
+            'mode': {
+                'type': 'string',
+                'enum': ['all', 'undefined', 'watch', 'get', 'blacklist', 'buddy'],
+                'default': 'all'},
+            'pass': {
+                'type': 'string'},
             'strip_dates': {
                 'type': 'boolean',
-                'default': False}
+                'default': False},
+            'user_agent': {
+                'type': 'string',
+                'default': ('Mozilla/5.0 (Windows NT 10.0; Win64; x64) '
+                            'AppleWebkit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36')}
         },
         'additionalProperties': False,
         'required': ['user_id'],
         'error_required': 'user_id is required'
     }
 
+    def __mode_string_to_int(self, mode_string):
+        if mode_string == 'buddy':
+            return 11
+        return self.schema['properties']['mode']['enum'].index(mode_string)
+
+    def __build_url(self, config):
+        base_url = self.anidb_url + 'animedb.pl?show=mywishlist&uid=%s' % config['user_id']
+        base_url = base_url +\
+            ('' if config['mode'] == 'all' else '&mode=%s' % self.__mode_string_to_int(config['mode']))
+        base_url = base_url + ('' if config['pass'] is None else '&pass=%s' % config['pass'])
+        return base_url
+
     @cached('anidb_list', persist='2 hours')
     def on_task_input(self, task, config):
         # Create entries by parsing AniDB wishlist page html using beautifulsoup
-        log.verbose('Retrieving AniDB list: mywishlist')
-        url = 'http://anidb.net/perl-bin/animedb.pl?show=mywishlist&uid=%s' % config['user_id']
-        log.debug('Requesting: %s' % url)
-
-        page = task.requests.get(url)
+        log.verbose('Retrieving AniDB list: mywishlist:%s' % config['mode'])
+        comp_link = self.__build_url(config)
+        log.debug('Requesting: %s', comp_link)
+        task_header = {
+            'User-Agent': config['user_agent']
+        }
+
+        page = task.requests.get(comp_link, headers=task_header)
```

Comment:
When using task.requests, the headers set via the `headers` plugin will automatically be used. It's totally fine to hard code a default user agent though

Reply:
This line should be wrapped in a try-except (RequestException) as well.